webapp/src/main/webapp/vue-apps/space-creation/components/SpaceCreationSettingsDrawer.vue

```diff
@@ -0,0 +1,180 @@
+          </template>
+        </v-radio>
+        <v-radio value="fewTemplates">
+          <template #label>
+            <span class="ms-1"> {{ $t('space.creation.instantiation.settingsDrawer.content.chooseTemplateChoice2') }}</span>
+          </template>
+        </v-radio>
+      </v-radio-group>
+      <v-autocomplete
+        v-if="spaceCreationTemplateChoice === 'fewTemplates'"
+        v-model="selectedTemplates"
+        :items="spaceTemplates"
+        :placeholder="$t('space.creation.instantiation.settingsDrawer.content.searchTemplatePlaceholder')"
+        item-text="name"
+        item-value="id"
+        class="mt-n2 mb-4 px-5 elevation-0 no-border"
+        hide-no-data
+        hide-selected
+        hide-details
+        outlined
+        multiple
+        return-object
+        dense />
+      <div v-if="spaceCreationTemplateChoice === 'fewTemplates'" class="px-4">
+        <v-chip
+          v-for="template in selectedTemplates"
+          :key="template.id"
+          class="my-1 me-3"
+          color="primary"
+          close
+          @click:close="removeTemplate(template)">
+          {{ template.name }}
+        </v-chip>
+      </div>
+    </template>
+    <template #footer>
+      <div class="d-flex align-center">
+        <v-btn
+          class="btn ms-auto me-2"
+          @click="reset">
+          {{ $t('space.creation.instantiation.cancel.button') }}
+        </v-btn>
+        <v-btn
+          :disabled="disabled"
+          :loading="loading"
+          class="btn primary"
+          @click="save">
+          {{ $t('space.creation.instantiation.apply.button') }}
+        </v-btn>
+      </div>
+    </template>
+  </exo-drawer>
+</template>
+<script>
+export default {
+  data: () => ({
+    drawer: false,
+    savedLabelTranslations: null,
+    language: eXo.env.portal.language,
+    spaceCreationTemplateChoice: null,
+    selectedTemplates: [],
+    disabled: false,
+    loading: false
+  }),
+  props: {
+    savedSettings: {
+      type: Object,
+      default: null
+    },
+    saveSettingsUrl: {
+      type: String,
+      default: null
+    }
+  },
+  computed: {
+    buttonLabel() {
+      return this.savedLabelTranslations || this.defaultLabel;
+    },
+    defaultLabel() {
+      return {[this.language]: this.$t('space.creation.instantiation.create.button')};
+    },
+  },
+  created() {
+    this.$root.$on('space-creation-settings-open', this.open);
+  },
+  beforeDestroy() {
+    this.$root.$off('space-creation-settings-open', this.open);
+  },
+  methods: {
+    open() {
+      this.restoreSavedSettings();
+      this.$refs.drawer.open();
+    },
+    close() {
+      this.$refs.drawer.close();
+    },
+    restoreSavedSettings() {
+      this.spaceTemplates = typeof this.savedSettings?.spaceTemplates === 'string' ? JSON.parse(this.savedSettings?.spaceTemplates) : this.savedSettings?.spaceTemplates;
```

Comment:
When can this attribute can be of type string ?

```suggestion
      this.spaceTemplates = this.savedSettings?.spaceTemplates;
```

Reply:
once the parameters have been saved, they will be presented as a string

Reply:
ok, can you change it as object at that time in order to ensure that it's all time of same type